src/test/java/bio/terra/integration/AzureIntegrationTest.java

```diff
@@ -199,6 +205,11 @@ public void teardown() throws Exception {
 
     dataRepoFixtures.resetConfig(steward);
 
+    if (snapshotAccessRequestIds != null) {
+      for (UUID snapshotAccessRequestId : snapshotAccessRequestIds) {
```

Comment:
Opt: Could be done in a functional way with `snapshotAccessRequestIds.forEach(snapshotAccessRequestId -> samFixtures.deleteSnapshotAccessRequest(steward, snapshotAccessRequestId))`.

You might be able to somehow make that null safe too, but that might be overkill.

That's also a change that could be made for `snapshotIds` below, but that's out of the PR changes, so up to you!